### DISCUSSION TITLE
General question about pieces

### DISCUSSION THREAD

**Discussion by isoctcolo** [Tauffer-Consulting/domino#245] [Q&A]
I checked the pieces repository and looks really simple.

My question is how would I implement for example to fetch data from an API.
For example I implement an API call to Salesforce which fetches me 40 different objects. 
How can this be easily done to configure which objects are shown in the UI and that the user can choose what objects he wants to export. 

Salesforce is jusgt an example, I want also to fetch Zoom, and some other Saas platforms to load it into PostgreSQL database.

And you think DBT (getdbt.com) can be integrated here ?

**Comment by luiztauffer** [comment]
Hi @isoctcolo 

Did you check our gallery of Pieces [here](https://domino-workflows.io/gallery)? Since Pieces can be installed from multiple different repositories, we’ll try our best to keep all the open source available Pieces listed over there. You can already find a couple of Pieces that make API calls and return results based on input arguments.

Definitely DBT could be integrated! We haven’t created any DBT specific Pieces just yet, so if you happen to create any Pieces for that purpose and decide to make them open source, let us know and we’ll be happy to list them in the gallery!
